src/main.rs

```diff
@@ -45,40 +45,208 @@ struct File {
-            .display()
-            .to_string(),
-        &database,
+    let migration_result = match migrate(files) {
+        Ok(migration_result) => migration_result,
+        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+    };
+
+    let uuid = match add_migration_result_to_db(
+        migration_result.0,
+        match String::from_utf8(migration_result.1) {
+            Ok(log) => log,
+            Err(e) => {
+                eprint!("Log was not utf8 encoded: {}", e);
+                return StatusCode::INTERNAL_SERVER_ERROR.into_response();
+            }
+        },
+        database,
     ) {
-        Ok(file_path) => file_path,
-        Err(_e) => {
-            return StatusCode::BAD_REQUEST.into_response();
-        }
+        Ok(uuid) => uuid,
+        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+    };
+    axum::response::Redirect::to(format!("{}/{}", redirect_path, uuid).as_str()).into_response()
+}
+
+fn read_from_db(uuid: String, database: &Connection) -> anyhow::Result<(String, String)> {
+    let mut select_stmt = database.prepare(
+        format!(
+            "SELECT file_path, log from {} WHERE uuid = (?1)",
+            TABLE_NAME
+        )
+        .as_str(),
+    )?;
+
+    let path_log = select_stmt.query_row([&uuid], |row| Ok((row.get(0)?, row.get(1)?)))?;
+    Ok(path_log)
+}
+
+fn generate_json(log: &str, files: Vec<File>) -> String {
+    let mut data = json::JsonValue::new_object();
+    data["log"] = log.into();
+    data["files"] = json::JsonValue::new_array();
+    for file in files {
+        let mut file_data = json::JsonValue::new_object();
+        file_data["fileName"] = file.file_name.into();
+        file_data["fileContent"] = file.file_content.into();
+        data["files"].push(file_data).unwrap();
+    }
+    data.dump()
+}
+
+fn file_arr_from_path(dir_path: String) -> Result<Vec<File>, anyhow::Error> {
+    let mut file_arr: Vec<File> = vec![];
+
+    let dir = match fs::read_dir(dir_path.clone() + "/NM-migrated/system-connections") {
+        Ok(dir) => dir,
+        Err(_e) => fs::read_dir(dir_path + "/NM-migrated")?,
+    };
+
+    for dir_entry in dir {
+        let path = dir_entry?.path();
+        let file_type = match path.extension() {
+            Some(file_type) => match file_type.to_str().unwrap() {
+                "xml" => FileType::Xml,
+                _ => FileType::Ifcfg,
+            },
+            None => {
+                return Err(anyhow::anyhow!("File extension was not recognized"));
+            }
+        };
+        let file_contents = std::fs::read(&path).unwrap();
+        file_arr.push(File {
+            file_content: String::from_utf8(file_contents).unwrap(),
+            file_name: path.file_name().unwrap().to_str().unwrap().to_owned(),
+            file_type,
+        });
+    }
+    Ok(file_arr)
+}
+
+async fn return_config_json(uuid: OriginalUri, State(shared_state): State<AppState>) -> Response {
```

Comment:
Getting the uuid from the uri path and stripping it is pretty hacky and unnecessary right? You should be able to get it as a param, that's what the `/:uuid` is for.